internal/handler/userHandler.go

```
@@ -56,17 +58,27 @@ func (h *Handler) Save(context *gin.Context) {
 	if err != nil {
 		logger.Error("Error while creating user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while mapping request to user: " + err.Error(),
+			Message:   "Error while mapping request to user",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
 		return
 	}
 	createdUser, err := h.service.SaveUser(context, user)
 	if err != nil {
+		var databaseConnectionErr *exceptions.DatabaseConnectionErr
+		if errors.As(err, &databaseConnectionErr) {
+			logger.Errorf("Error while connecting to database. %v", err)
+			errorResponse := response.ErrorDto{
+				Message:   "Something went wrong",
+				Timestamp: time.Now(),
+			}
+			context.IndentedJSON(http.StatusInternalServerError, errorResponse)
+			return
+		}
 		logger.Error("Error while persisting user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while creating user. " + err.Error(),
+			Message:   "Error while creating user.",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
```

There's no need to check for a database specific error. If the save call returned an error there's no way we can recover so we should log the error and return a 500 status with a generic error message.

---

But my method returns an specific error for databaseConnection and for userNotFound, for example, I need to check which error im receiving, so I can return the correct status code and message

---

Since GO is a new language the standards have evolved much more rapidly then Java over the last 10 years. If the Mongo driver was being rewritten from scratch they would likely not return an error if a document was not found. See the following section of code, our standards at Wex are to treat no documents being found as an expected case, not an error.
https://github.com/wexinc/ps-tag-dev/blob/290a44beab36b0167ff00a6e545b6a9ca6e85772/pkg/store/mongodriver/store.go#L404-L410
Go does not have an equivalent to Optional in Java. Because of this we can treat it like Java before Optional where a null value means no records were found. Since an expected case is a user querying a missing user not getting a result shouldn't be an error.

---

I don't know if this is a best practice, since it's an error from user-side, we should return an error. It will be more easier to understand what is happening, given that  we will threat with something like this: if user == nil {logError}
The difference is just that I'm getting an error instead of verifying user nil.
But as you mentioned that is what we do here, I can change it.